### ISSUE TITLE
Setting goos, goarch causes cgo setting to be ignored

### ISSUE THREAD

**Issue by rohansingh** [bazelbuild/rules_go#2539]
### What version of rules_go are you using?

v023.0

### What version of gazelle are you using?

v0.21.0

### What version of Bazel are you using?

3.1.0

### Does this issue reproduce with the latest releases of all the above?

Need to confirm but pretty sure.

### What operating system and processor architecture are you using?

Linux, amd64

### What did you do?

For a `go_binary` target that depends on a cgo-enabled library, set `goos`, `goarch`, and `cgo`, like this:

```
go_binary(
    name = "my-server",
    goos = "linux",
    goarch = "amd64",
    cgo = True,
    embed = [":go_default_library"],
    visibility = ["//visibility:public"],
)
```

### What did you expect to see?

Successful build, or a descriptive error telling me that this configuration is unsupported.

### What did you see instead?

Build failure due to undefined symbols. Running with `--sandbox_debug` shows `CGO_ENABLED=0`. Removing `goos` and `goarch` fixes the issue, but as long as they are present, the `cgo` setting has no effect.

**Comment by jayconrod** [comment]
`cgo = True` indicates that a package may have cgo code, and the builder may need to do some extra work to process it. However, it doesn't force cgo to be used if the toolchain isn't configured to use it.

I think what's missing here is `pure = "off"`. Does adding that work for you?

Marking this as a documentation bug, since the reference documentation isn't very clear on this.

**Comment by rohansingh** [comment]
Sorry for the late response here. Explicitly setting `pure = "off"` did indeed solve the issue.

**Comment by rohansingh** [comment]
This may be a 101-level question, but doesn't cgo imply that pure should be off?

**Comment by jayconrod** [comment]
No, `cgo = True` just indicates that the package contains cgo code. If cgo is enabled, the rules may need to declare extra output files and actions. The rules can't read the files during analysis, so it needs to be declared ahead of time. If cgo is disabled, cgo files will be ignored, and the package may provide a pure Go implementation (with `// +build !cgo`).

It's kind of like testing whether the `CgoFiles` list is empty in `go list -json` output.

**Comment by rohansingh** [comment]
Got it. Thanks for the explanation!
